mypy/nodes.py

```diff
@@ -1984,6 +1994,8 @@ class OpExpr(Expression):
     right_always: bool
     # Per static analysis only: Is the right side unreachable?
     right_unreachable: bool
+    # Used for expressions that represent a type "X | Y" in some contexts
+    analyzed: TypeAliasExpr | None
```

Comment:
This should now be handled in various visitors. Three cases come to my mind: First, `treetransform.py` this is needed so that this error will not re-appear in a test case like this
```python
T = TypeVar("T", int, str)
def foo(x: T) -> T:
    A = type[int] | str
    return x
```
Second, `aststrip.py`, be sure that when you switch imported names from types to variables, you do get an error about missing `__or__` on update. Third, `semanal_typeargs.py` (uses `MixedTraverserVisitor`), since we should not carry malformed instances around (with number of type args), they may cause crashes, add a test just in case with a malformed instance in `|` alias.

And in general adding in to the basic `TraverserVisitor` is a good idea. Maybe just grep for `def visit_index_expr(` and see where we use `analyzed`.

Reply:
Very good points! I'll fix these.